include/bit7z/bitindicesview.hpp

```diff
@@ -19,97 +20,14 @@
 #include <vector>
 
 namespace bit7z {
+
 using IndicesVector = std::vector< std::uint32_t >;
 
 template< std::size_t N >
 using IndicesArray = std::array< std::uint32_t, N >;
 
-// NOLINTBEGIN(*-explicit-conversions, *-avoid-c-arrays, *-pro-bounds-pointer-arithmetic)
-class BitIndicesView final {
```

Comment:
before we were talking about ByteSpan, but I noticed this class, which is exactly what was needed, except for the generalisation, so I extracted the general part.

I also named it (in line with what is here) BitView<T>

but of course I can name it ByteSpan as we previously spoke